src/sysfiles.c

```diff
@@ -440,6 +447,119 @@ Int SyLoadModule( const Char * name, InitInfoFunc * func )
 
 #endif
 
+/****************************************************************************
+**
+*F * * * * * * * * * * finding location of executable * * * * * * * * * * * *
+*/
+
+/****************************************************************************
+** The function 'find_yourself' is based on code (C) 2015 Mark Whitis, under
+** the MIT License : http://stackoverflow.com/a/34271901/928031
+*/
+
+void find_yourself(const char * argv0, char * result, size_t size_of_result)
```

Comment:
The `find_yourself` method is cut+pasted from stackoverflow, complete with long variables names, etc.

I could make it "more GAPish", but I'd prefer to leave it as is as much as possible, because I assume the code is well tested (or people would have hopefully complained on stackoverflow), and I can't test every OS it runs on easily.